cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -1000,16 +1010,16 @@ struct GroupedMinMaxImpl : public GroupedAggregator {
 
     mins_ = BufferBuilder(ctx->memory_pool());
     maxes_ = BufferBuilder(ctx->memory_pool());
-    has_values_ = BufferBuilder(ctx->memory_pool());
-    has_nulls_ = BufferBuilder(ctx->memory_pool());
+    has_values_ = TypedBufferBuilder<bool>(ctx->memory_pool());
+    has_nulls_ = TypedBufferBuilder<bool>(ctx->memory_pool());
 
     GetImpl get_impl;
     RETURN_NOT_OK(VisitTypeInline(*input_type, &get_impl));
 
     consume_impl_ = std::move(get_impl.consume_impl);
     resize_min_impl_ = std::move(get_impl.resize_min_impl);
     resize_max_impl_ = std::move(get_impl.resize_max_impl);
-    resize_bitmap_impl_ = MakeResizeImpl(false);
+    resize_bitmap_impl_ = MakeResizeImplForBitmap(false);
```

Comment:
I think it's unnecessary to have a separate `std::function` for resizing the bitmaps, let's just linline the calls to TypedBufferBuilder::Append inside the MaybeReserve call

Reply:
Sure, now `resize_bitmap_impl_` is replaced with `Append` in `MaybeReserve`